sdk/iot/README.md

```diff
@@ -67,11 +67,11 @@ void my_property_func()
 {
   //Allocate a span to put the properties
   uint8_t property_buffer[64];
-  az_span property_span = az_span_init(property_buffer, 0, sizeof(property_buffer));
+  az_span property_span = az_span_init(property_buffer, sizeof(property_buffer));
   
   //Initialize the property struct with the span
   az_iot_hub_client_properties props;
-  az_iot_hub_client_properties_init(&props, property_span);
+  az_iot_hub_client_properties_init(&props, property_span, 0);

```

Comment:
@CIPop - please take a look at this as it affects the public API surface of `az_iot_hub_client_properties` and its uses.

We have two options here, that I can think of:
1) Disallow the ability for someone to pass in a pre-initialized buffer containing properties at init time and have the user call append.
   - We can consider adding an `append_property` overload that takes a span, if that is useful.
2) Add a `written_length` parameter to init that tells us if the buffer has some initial data.

I picked (2), but let me know what you think the best approach is here and what is most suitable for end users. If you prefer some other design, please let me know.

Reply:
I like 2